src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -968,49 +961,46 @@ ORDER BY
                             )
                             SELECT ISNULL(MIN(ResourceSurrogateId), 0), ISNULL(MAX(ResourceSurrogateId), 0), COUNT(*) FROM A
```

Comment:
SELECT isnull(min(ResourceSurrogateId), 0), isnull(max(ResourceSurrogateId), 0), count(*)
  FROM (SELECT TOP (@p3) ResourceSurrogateId 
          FROM dbo.Resource 
          WHERE ResourceTypeId = @p1 
            AND IsHistory = 0 
            AND IsDeleted = 0
            AND ResourceSurrogateId > @p2
            AND (SearchParamHash != @p0 OR SearchParamHash IS NULL)
          ORDER BY
               ResourceSurrogateId
       ) A

If you like it better, then replace.